docs/getting_started/installation_steps.md

```diff
@@ -0,0 +1,306 @@
+
+**Download the latest release compiled for your system from [GitHub release assets](https://github.com/gohornet/hornet/releases), for ex:**
+
+```bash
+curl -LO https://github.com/gohornet/hornet/releases/download/v0.6.0/HORNET-0.6.0_Linux_x86_64.tar.gz
+```
+Some navigation hints:
+* `HORNET-X.Y.Z_Linux_x86_64.tar.gz`: standard 64-bit-linux-based executable, such as Ubuntu, Debian, etc.
+* `HORNET-X.Y.Z_Linux_arm64.tar.gz`: executable for Raspberry Pi 4
+* `HORNET-X.Y.Z_Windows_x86_64.zip`: executable for Windows 10-64-bit-based systems
+* `HORNET-X.Y.Z_macOS_x86_64.tar.gz`: executable for macOS
+
+**Extract the files in a folder of your choice (for ex. `/opt` on Linux), for ex:**
+```bash
+tar -xf HORNET-0.6.0_Linux_x86_64.tar.gz
+```
+* Once extracted, you get a main executable file, for example `hornet` for linux, or `hornet.exe` for Windows
+* There are also sample [configuration](../post_installation/post_installation.md) JSON files available in the archive (tar or zip)
+
+**Run Hornet using `--help` to get all executable-related arguments:**
+```bash
+./hornet --help
+```
+
+*Also double check that you have version 0.6.0+ deployed:*
+```bash
+./hornet --version
+```
+
+**Run Hornet using default settings:**
+```bash
+./hornet
+```
+
+Using this method, you have to make sure the executable runs in a daemon mode using for example `systemd`.
+
+> Please note: Hornet uses an in-memory cache and so it is necessary to provide a grace period while shutting it down (at least 200 seconds) in order to save all data to the underlying persistent storage.
+
+See more details on how to configure Hornet under the [post installation](../post_installation/post_installation.md) chapter.
+
+### Example of systemd unit file
+Assuming the Hornet executable is extracted to `/opt/hornet` together with configuration files, please find the following example of a `systemd` unit file:
+
+```plaintext
+[Unit]
+Description=Hornet
+Wants=network-online.target
+After=network-online.target
+
+[Service]
+StandardOutput=syslog
+StandardError=syslog
+SyslogIdentifier=hornet
+PrivateDevices=yes
+PrivateTmp=yes
+ProtectSystem=full
+ProtectHome=yes
+
+User=hornet
+WorkingDirectory=/opt/hornet
+TimeoutSec=1200
+Restart=always
+ExecStart=/opt/hornet/hornet
+
+[Install]
+WantedBy=multi-user.target
+```
+
+
+----------------
+
+
+
+# Build from source
+This method is considered a bit advanced for production use since you have to usually prepare a system environment in order to run the given executable as a service (in a daemon mode) via `systemd` or `supervisord`.
+
+*Ubuntu/Debian*
+
+*There is usually quite old version of Go language in standard `apt` repositories available, and so it is better to add `golang-backports` PPA to get the latest version.*
+
+**Install dependencies: Go, git and build-essentials:**
+```bash
+sudo add-apt-repository ppa:longsleep/golang-backports
+sudo apt update
+sudo apt install golang-go git build-essential
```

Comment:
```suggestion
This method is considered a bit advanced for production use since you usually have to prepare a system environment in order to run the given executable as a service (in a daemon mode) via `systemd` or `supervisord`.

*Ubuntu/Debian*

*There is usually quite old version of Go language in standard `apt` repositories available, so it is better to add `golang-backports` PPA to get the latest version.*

**Install dependencies: Go, git and build-essentials:**
```bash
sudo add-apt-repository ppa:longsleep/golang-backports
sudo apt update
sudo apt install golang-go git build-essential
```